src/Zipper.elm

```diff
@@ -0,0 +1,81 @@
+module Zipper exposing
+    ( Zipper
+    , getCurrent
+    , makeZipper
+    , moveBackwards
+    , moveBackwardsTo
+    , moveBackwardsWhile
+    , moveForward
+    , moveForwardTo
+    , moveForwardWhile
+    )
+
+import Debug
+import List exposing (append, head, isEmpty, tail)
+import List.Extra as ListE
+import Maybe exposing (withDefault)
+
+
+type alias Zipper a =
+    { previous : List a
+    , current : a
+    , next : List a
+    }
+
+
+makeZipper : List a -> a -> List a -> Maybe (Zipper a)
+makeZipper previous current next =
+    Just { previous = previous, current = current, next = next }
+
+
+moveForward : Zipper a -> Maybe (Zipper a)
+moveForward zipper =
+    Just <|
+        if isEmpty zipper.previous then
+            --TODO: figure out whether it should actually be returning "Nothing"
+            zipper
+
+        else
+            { previous = withDefault [] <| ListE.init zipper.previous
+            , current = withDefault zipper.current <| ListE.last zipper.previous
```

Comment:
You don't need `withDefault` here - you already know you this default is never used, because you're in the `else` branch so `zipper.previous` isn't empty.
This is a hint you can implement this function better 🙂.

Hint: use pattern matching.